variable/include/scipp/variable/variable.tcc

```diff
@@ -60,4 +61,20 @@ template <class T> ElementArrayView<T> Variable::variances() {
   Variable::variances() const;                                                 \
   template SCIPP_EXPORT ElementArrayView<__VA_ARGS__> Variable::variances();
 
+template <class T> std::string Formatter<T>::format(const Variable &var) const {
+  return array_to_string(var.template values<T>());
+}
+
+/// Insert classes into formatting registry. The objects themselves do nothing,
+/// but the constructor call with comma operator does the insertion. Calling
+/// this is required for formatting all but basic builtin types.
+#define REGISTER_FORMATTER(name, ...)                                          \
+  namespace {                                                                  \
+  auto register_##name(                                                        \
+      (variable::formatterRegistry().emplace(                                  \
+           dtype<__VA_ARGS__>,                                                 \
+           std::make_unique<variable::Formatter<__VA_ARGS__>>()),              \
+       0));                                                                    \
+  }
```

Comment:
Is this safe? Do we still initialise other things statically?

Also, why use variable arguments? Looks like a single one is enough.

Reply:
Which part may not be safe? Note that this is not new, just defining a macro for what was done directly previously.

The variable arguments are required for types with spaces such as `std::pair<A, B>`.

Reply:
> Which part may not be safe? Note that this is not new, just defining a macro for what was done directly previously.

I misread `register_python_types` etc. as function definitions. So I thought there was some order to calls to the registry. But never mind.

> The variable arguments are required for types with spaces such as `std::pair<A, B>`.

Right